src/OCServo.cpp

```diff
@@ -0,0 +1,481 @@
+    byte checksum = 0;
+    for (byte i = 2; i < size; i++)
+    {
+        checksum += data[i];
+    }
+    return ~(checksum & 0xFF);
+}
+
+byte OCServo::baudRateToByte(long baudrate) {
+    byte value = 0x00;
+
+    switch (baudrate) {
+        case 500000:
+            value = OCS_BAUDRATE_500K;
+            break;
+        case 250000:
+            value = OCS_BAUDRATE_250K;
+            break;
+        case 128000:
+            value = OCS_BAUDRATE_128K;
+            break;
+        case 115200:
+            value = OCS_BAUDRATE_115200;
+            break;
+        case 76800:
+            value = OCS_BAUDRATE_76800;
+            break;
+        case 57600:
+            value = OCS_BAUDRATE_57600;
+            break;
+        case 38400:
+            value = OCS_BAUDRATE_38400;
+            break;
+        default:
+            // Fallthrough to default 1M
+        case 1000000:
+            value = OCS_BAUDRATE_1M;
+            break;
+    }
+
+    return value;
+}
+
+long OCServo::byteToBaudRate(byte value) {
+    long result;
+
+    switch(value) {
+        case OCS_BAUDRATE_1M:
+            result = 1000000;
+            break;
+        case OCS_BAUDRATE_500K:
+            result = 500000;
+            break;
+        case OCS_BAUDRATE_250K:
+            result = 250000;
+            break;
+        case OCS_BAUDRATE_128K:
+            result = 128000;
+            break;
+        case OCS_BAUDRATE_115200:
+            result = 115200;
+            break;
+        case OCS_BAUDRATE_76800:
+            result = 76800;
+            break;
+        case OCS_BAUDRATE_57600:
+            result = 57600;
+            break;
+        case OCS_BAUDRATE_38400:
+            result = 38400;
+            break;
+        default:
+            result = -1;
+            break;
+    }
+
+    return result;
+}
+
+OCSResponse OCServo::bytesToResponse(byte *data, int size) {
+    OCSResponse response;
+
+    response.prefix[0] = *(data);
+    response.prefix[1] = *(data+1);
+    response.id = data[2];
+    response.length = data[3];
+    response.instruction = data[4];
+    response.numberOfParameters = response.length - 2;
+    for(int i = 0; i < response.numberOfParameters; i++) {
+        response.parameters[i] = data[5+i];
+    }
+    response.checksum = data[response.numberOfParameters + 5];
+
+    return response;
+}
+
+OCSResponse OCServo::readResponse() {
+    byte response[23];  // 16 params + 7 other fields
+    for (int x = 0; x < 23; x++) response[x] = 0x00; // Clear response array (just in case)
+    int i = 0;
+
+    unsigned long last_check_time = millis();
+    while(true) {
+        if(serial->available() > 0) {
+            response[i] = serial->read();
+            i++;
+            last_check_time = millis();
+        }
+        else if(millis() > last_check_time + 4) {
+            break;
+        }
+    }
+
+    OCSResponse result = this->bytesToResponse(response, i);
+    return result;
+}
```

Comment:
Also buffer overflow, and more logical check for timeout
```suggestion
OCSResponse OCServo::readResponse() const {
    uint8_t response[23] = {};  // 16 params + 7 other fields
    size_t i = 0;

    unsigned long lastCheckTime = millis();
    while (true) {
        if (serial.available() > 0) {
            response[i++] = static_cast<uint8_t>(serial.read());
            lastCheckTime = millis();
            if (i >= sizeof(response)) break; // Prevent overflow
        } else if (millis() - lastCheckTime > 4) {
            break; // Timeout: no new data within 4 ms
        }
    }

    return bytesToResponse(response, i);
}
```